scope.go

```diff
@@ -82,11 +91,14 @@ type Scope struct {
 
 func newScope() *Scope {
 	s := &Scope{
-		providers: make(map[key][]*constructorNode),
-		values:    make(map[key]reflect.Value),
-		groups:    make(map[key][]reflect.Value),
-		invokerFn: defaultInvoker,
-		rand:      rand.New(rand.NewSource(time.Now().UnixNano())),
+		providers:       make(map[key][]*constructorNode),
+		decorators:      make(map[key][]*decoratorNode),
+		values:          make(map[key]reflect.Value),
+		decoratedValues: make(map[key]reflect.Value),
+		groups:          make(map[key][]reflect.Value),
+		decoratedGroups: make(map[key][]reflect.Value),
+		invokerFn:       defaultInvoker,
+		rand:            rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
The randomness here is here just so that people don't rely on the ordering of grouped values when they use Dig; I don't think it not being "random enough" would matter that much, but we could change it. Thoughts?

Reply:
For security purposes, crypto/rand is indeed the thing to use.
To shuffle the Dig value groups, pseudo-randomness from math/rand is enough.
(Slightly related: crypto/rand will be really slow for this.)